proposals/22-03-value-in-the-structured-data.md

```diff
@@ -0,0 +1,511 @@
+          }
+        }
+      }
+```
+
+```yaml
+# overlay/kustomization.yaml
+resources:
+- ../base
+configMapGenerator:
+- name: demo
+  option:
+    valueMergeFormat: json
+  literals:
+  - config.json: |-
+      {
+        "config": {
+          "hostname": "www.example.com",
+          "parameter": {
+            "baz": "qux"
+          }
+        }
+      }
+```
+
+#### Result
+```yaml
+apiVersion: v1
+data:
+  config.json: |-
+    {
+      "config": {
+        "loglevel": debug,
+        "hostname": "www.example.com",
+        "parameter": {
+          "foo": "bar",
+          "baz": "qux"
+        }
+      }
+    }
+kind: ConfigMap
+metadata:
+  name: demo-xxxxxxxxxx  # name suffix hash
+```
+
+
+#### Story 3
+
+Scenario summary: Replacement the value inside for yaml in the configMap.
+<!--
+A walkthrough of what it will look like for a user to take advantage of the new feature.
+Include the the steps the user will take and samples of the commands they'll run
+and config they'll use.
+-->
+
+A few applications require to use yaml format config file, and some major cloudnative applications are using that.(ex, Prometheus,AlertManager)\
+A value you want to overlay in yaml is usually into a nested yaml structure. So if you can overlay value inside yaml, you won't need to copy a whole yaml file.
+
+```yaml
+## source
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: environment-config
+data:
+  env: dev
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: prometheus-config
+data:
+  prometheus.yml: |-
+    global:
+      external_labels:
+        prometheus_env: TARGET_ENVIROMENT
+    scrape_configs:
+      - job_name: "prometheus"
+        static_configs:
+          - targets: ["localhost:9090"]
+```
+
+```yaml
+## replacement
+replacements:
+- source:
+    kind: ConfigMap
+    name: environment-config
+    fieldPath: data.env
+  targets:
+  - select:
+      kind: ConfigMap
+      name: prometheus-config
+    fieldPaths:
+    - prometheus\.yml
+    options:
+      format: 'yaml'
+      formatPath: '/global/external_labels/prometheus_env'
+```
+
+```yaml
+## expected
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: environment-config
+data:
+  env: dev
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: prometheus-config
+data:
+  prometheus.yml: |-
+    global:
+      external_labels:
+        prometheus_env: dev
+    scrape_configs:
+      - job_name: "prometheus"
+        static_configs:
+          - targets: ["localhost:9090"]
+```
+
+#### Story 4
+
+Scenario summary: Replacement the value inside for json in the annotations.
+<!--
+A walkthrough of what it will look like for a user to take advantage of the new feature.
+Include the the steps the user will take and samples of the commands they'll run
+and config they'll use.
+-->
+
+A few times, an application on your cluster requires to set json format config for the *Annotations* on kubernetes yaml resources. We need to overlay in this position value.
+
+##### Example
```

Comment:
This example seems to be missing the Kustomization and overlay portions

Reply:
I add it.

https://github.com/kubernetes-sigs/kustomize/pull/4558/commits/83c7492d569dabf15050cfdba254f2717c0ac332#diff-2b5a81ecf356afb37e2653ac1e2daf0b6af6cd126aec835df3c25b2a8be29389R437-R487